src/Tests/ObjectLifetimeTests/Tests/ObjectLifetimeTests.cs

```diff
@@ -0,0 +1,1483 @@
+
+            _asyncQueue
+                .CallFromUIThread(() =>
+                {
+                    // Create cycle.
+                    CustomControl parent = new CustomControl();
+                    Button child = new Button();
+                    parent.Content = child;
+
+                    BindingOperations.SetBinding(
+                        child,
+                        FrameworkElement.TagProperty,
+                        new Binding() { Source = parent, Path = new PropertyPath("Tag") }
+                    );
+
+                    mainCanvas.Children.Append(parent);
+
+                    weakRef1 = new WeakReference(parent);
+                    weakRef2 = new WeakReference(child);
+                })
+                .CallFromUIThread(() =>
+                {
+                    Verify(weakRef1.IsAlive, "parent died");
+
+                    // Verify child is collected.
+                    GC_ForceCollect();
+
+                    Verify(weakRef1.IsAlive, "parent died");
+
+                    // Remove references to parent.
+                    mainCanvas.Children.RemoveAt(mainCanvas.Children.Count - 1);
+                })
+                .CallFromUIThread(() =>
+                {
+                    GC_ForceCollect();
+
+                    Verify(!weakRef1.IsAlive, "parent is alive");
+                    Verify(!weakRef2.IsAlive, "child is alive");
+                });
+
+            _asyncQueue.Run();
+        }
+
+        //
+        // Cycle:
+        // Grid
+        // => Binding(CustomDependencyObject.AttachedObject = Button.Tag)
+        // => Button.Tag = Grid
+        //
+        [MethodImplAttribute(MethodImplOptions.NoOptimization | MethodImplOptions.NoInlining)]
+        public void CycleTest9()
+        {
+            WeakReference weakRef1 = null;
+
+            _asyncQueue
+                .CallFromUIThread(() =>
+                {
+                    // Create cycle.
+                    Grid grid = new Grid();
+                    Button button = new Button();
+                    BindingOperations.SetBinding(grid, CustomDependencyObject.AttachedObjectProperty, new Binding() { Source = button, Path = new PropertyPath("Tag") });
+                    button.Tag = grid;
+
+                    // Verify collection.
+                    weakRef1 = new WeakReference(grid);
+                    MakeNull(ref grid);
+                    MakeNull(ref button);
+                })
+                .CallFromUIThread(() =>
+                {
+                    GC_ForceCollect();
+
+                    Verify(!weakRef1.IsAlive, "grid is alive");
+                });
+
+            _asyncQueue.Run();
+        }
+
+        //
+        // Cycle:
+        // CustomDependencyObject
+        // => Binding(CustomDependencyObject.AttachedObject = Button.Tag)
+        // => Button.Tag = CustomDependencyObject
+        //
+        //[Disabled("595913", "CodeDefect")] // resolved Won't Fix
```

Comment:
Remove?  I assume this works now?

Reply:
Still a failing test, the absence of `[TestMethod]`  made it not run.